src/main/scala/is/hail/rvd/OrderedRVD.scala

```diff
@@ -21,29 +21,19 @@ class OrderedRVD private(
   self =>
   def rowType: TStruct = typ.rowType
 
-  def insert[PC](newContext: () => PC)(typeToInsert: Type,
-    path: List[String],
-    // rv argument to add is the entire row
-    add: (PC, RegionValue, RegionValueBuilder) => Unit): OrderedRVD = {
-    val localTyp = typ
-
-    val (insTyp, inserter) = typ.insert(typeToInsert, path)
-    OrderedRVD(insTyp,
-      partitioner,
-      rdd.mapPartitions { it =>
-        val c = newContext()
-        val rv2b = new RegionValueBuilder()
-        val rv2 = RegionValue()
+  // should be totally generic, permitting any number of keys, but that requires more work
+  def downcastToPK(): OrderedRVD = {
+    val newType = new OrderedRVType(partitionKey = typ.partitionKey,
+      key = typ.partitionKey,
+      rowType = rowType)
+    OrderedRVD(newType, partitioner, rdd)
```

Comment:
I see what you're doing here, changing the ORVDType, but it feels like calling this a case is confusing.

I think you need to change the partitioner kType here, too.  We should probably have an assertion for this in the OrderedRVD constructor.

Reply:
I sorta feel like we have the wrong primitives -- maybe orderedJoinDistinct should take the join key, and check that it's a shared prefix between the PK and K for both?

Reply:
is it OK if I leave it broken-ish for now? I'm worried I break more stuff trying to fix it.